### PR TITLE
Removed calls to ADMaterial

### PR DESCRIPTION
**Pull request by tophmatthews**
ADMaterial.h can be used interchangeably with Material.h, and leads to confusion on when to use which. Really, all calls to ADMaterial can just be Material (I think), so make it so.

closes #19091

### CODE REVIEW

**Comment by GiudGiud** [comment]
I think the idea behind having both was that if there ever needed to be differences between material and ADMaterial, the right headers would already be included.

**Comment by lindsayad** [comment]
`ADMaterial` used to be a concrete class but now it is not after we removed `ComputeStage`. I don't anticipate it ever coming back. We left `ADMaterial` around as a `typedef` for backwards compatibility. It can be safely removed from your code in favor of `Material` everywhere

**Comment by dschwen** [comment]
This is something that can and should be done in downstream apps first, then we can completely remove anything `ADMaterial` in MOOSE.

**Comment by GiudGiud** [comment]
There's an ongoing Pronghorn failure. It is being addressed (fix should be merged once we have people s approval)
See herd-ccb channel for updates on civet

**Comment by loganharbour** [comment]
@tophmatthews can't you just include an empty header for now that includes `Material.h`?

**Comment by lindsayad** [comment]
> How can it be deprecated?

We now require C++14, so [this method](https://stackoverflow.com/a/68370938/4493669) could be used

**Comment by tophmatthews** [comment]
I didn't take out downstream uses of `ADMaterial`, but this should be okay to go regardless.

**Review by lindsayad — APPROVED** [review]
thanks @tophmatthews this is a nice cleanup

**Comment by lindsayad** [comment]
I'll merge when next is green again (and you can ping me if I forget)

**Pull request merged**
